ecosystem/platform/server/app/controllers/it2_profiles_controller.rb

```diff
@@ -90,9 +90,34 @@ def update
 
   private
 
+  def create_node_checker(node_checker_base_url, hostname, metrics_port, http_api_port, noise_port)
+    if Flipper.enabled?(:node_health_checker)
+      NodeHelper::NodeChecker.new(node_checker_base_url, hostname, metrics_port, http_api_port, noise_port)
+    else
+      NodeHelper::NodeVerifier.new(hostname, metrics_port, http_api_port)
+    end
+  end
+
   # @param [NodeHelper::NodeChecker] node_verifier
   # @return [Array<VerifyResult>]
   def validate_node(node_verifier, do_location: false)
+    return validate_node_nhc(node_verifier, do_location:) if Flipper.enabled?(:node_health_checker)
```

Comment:
lol @ the race condition this could enable (I'm fine with it)

Reply:
I can change it to `if node_verifier.is_a? NodeHelper::NodeChecker`.